llvm/test/CodeGen/AArch64/vararg.ll

```diff
@@ -0,0 +1,569 @@
+; CHECK-SD-NEXT:    stur x5, [x29, #-32]
+; CHECK-SD-NEXT:    stur x4, [x29, #-40]
+; CHECK-SD-NEXT:    stur x3, [x29, #-48]
+; CHECK-SD-NEXT:    stur x2, [x29, #-56]
+; CHECK-SD-NEXT:    stur x1, [x29, #-64]
+; CHECK-SD-NEXT:    mov w8, #-112 // =0xffffff90
+; CHECK-SD-NEXT:    str w8, [x29, #20]
+; CHECK-SD-NEXT:    mov w8, #-56 // =0xffffffc8
+; CHECK-SD-NEXT:    str w8, [x29, #16]
+; CHECK-SD-NEXT:    add x8, x29, #16
+; CHECK-SD-NEXT:    stur x8, [x29, #-8]
+; CHECK-SD-NEXT:    mov x8, sp
+; CHECK-SD-NEXT:    add x8, x8, #112
+; CHECK-SD-NEXT:    str x8, [x29, #8]
+; CHECK-SD-NEXT:    sub x8, x29, #64
+; CHECK-SD-NEXT:    add x8, x8, #56
+; CHECK-SD-NEXT:    str x8, [x29]
+; CHECK-SD-NEXT:    mov w8, #1 // =0x1
+; CHECK-SD-NEXT:    mov w0, w8
+; CHECK-SD-NEXT:    .cfi_def_cfa wsp, 192
+; CHECK-SD-NEXT:    ldp x29, x30, [sp, #176] // 16-byte Folded Reload
+; CHECK-SD-NEXT:    add sp, sp, #192
+; CHECK-SD-NEXT:    .cfi_def_cfa_offset 0
+; CHECK-SD-NEXT:    .cfi_restore w30
+; CHECK-SD-NEXT:    .cfi_restore w29
+; CHECK-SD-NEXT:    ret
+;
+; CHECK-GI-LABEL: gpr1_fpr1:
+; CHECK-GI:       // %bb.0: // %entry
+; CHECK-GI-NEXT:    sub sp, sp, #208
+; CHECK-GI-NEXT:    .cfi_def_cfa_offset 208
+; CHECK-GI-NEXT:    stp x29, x30, [sp, #192] // 16-byte Folded Spill
+; CHECK-GI-NEXT:    add x29, sp, #192
+; CHECK-GI-NEXT:    .cfi_def_cfa w29, 16
+; CHECK-GI-NEXT:    .cfi_offset w30, -8
+; CHECK-GI-NEXT:    .cfi_offset w29, -16
+; CHECK-GI-NEXT:    stur x1, [x29, #-56]
+; CHECK-GI-NEXT:    stur x2, [x29, #-48]
+; CHECK-GI-NEXT:    stur x3, [x29, #-40]
+; CHECK-GI-NEXT:    stur x4, [x29, #-32]
+; CHECK-GI-NEXT:    stur x5, [x29, #-24]
+; CHECK-GI-NEXT:    stur x6, [x29, #-16]
+; CHECK-GI-NEXT:    stur x7, [x29, #-8]
+; CHECK-GI-NEXT:    str q1, [sp, #16]
+; CHECK-GI-NEXT:    str q2, [sp, #32]
+; CHECK-GI-NEXT:    str q3, [sp, #48]
+; CHECK-GI-NEXT:    str q4, [sp, #64]
+; CHECK-GI-NEXT:    str q5, [sp, #80]
+; CHECK-GI-NEXT:    str q6, [sp, #96]
+; CHECK-GI-NEXT:    str q7, [sp, #112]
+; CHECK-GI-NEXT:    add x9, sp, #8
+; CHECK-GI-NEXT:    add x8, x29, #16
+; CHECK-GI-NEXT:    str x8, [x9]
+; CHECK-GI-NEXT:    add x8, x29, #0
+; CHECK-GI-NEXT:    str x8, [x9, #8]
+; CHECK-GI-NEXT:    add x8, sp, #128
+; CHECK-GI-NEXT:    str x8, [x9, #16]
+; CHECK-GI-NEXT:    mov w8, #-56 // =0xffffffc8
+; CHECK-GI-NEXT:    str w8, [x9, #24]
+; CHECK-GI-NEXT:    mov w8, #-112 // =0xffffff90
+; CHECK-GI-NEXT:    str w8, [x9, #28]
+; CHECK-GI-NEXT:    mov w8, #1 // =0x1
+; CHECK-GI-NEXT:    mov w0, w8
+; CHECK-GI-NEXT:    .cfi_def_cfa wsp, 208
+; CHECK-GI-NEXT:    ldp x29, x30, [sp, #192] // 16-byte Folded Reload
+; CHECK-GI-NEXT:    add sp, sp, #208
+; CHECK-GI-NEXT:    .cfi_def_cfa_offset 0
+; CHECK-GI-NEXT:    .cfi_restore w30
+; CHECK-GI-NEXT:    .cfi_restore w29
+; CHECK-GI-NEXT:    ret
+entry:
+  %g = alloca ptr, align 4
+  call void @llvm.va_start(ptr %g)
+  ret i64 1
+}
+
+; A real program case
+define float @vararg_program(ptr %a, ...) #0 {
```

Comment:
I think it would be a good idea to add a comment briefly describing what the test does in more detail given its length.

Reply:
Maybe we should remove this case instaed, because it is rather long, includes a lot of things out of the scope of `va_start`, and is hard to know if it actually work. We can craft simpler ones if it's needed.

Reply:
This function should be broken up to test one case per function, with no control flow

Reply:
> This function should be broken up to test one case per function, with no control flow

Then it's essentially duplicating `define i64 @vararg(...) #0`.  Because this PR only modifies va_start and other blocks does not use `va_start`. 

I think we can remove it instead.